### ISSUE TITLE
[Feature Request] Atomic get and delete of normal key-values

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ashish21590/redis#160]
```
One common usage pattern we are attempting to implement is loading data
into a Redis store for aggregation using increment. If we process a key's
worth of data, then delete the key, the key might've been incremented
before we can delete it but after we've processed the value -- a classic
data race. What we'd like to do is grab the key's value and have the key
deleted when the value is returned atomically.
```

Original issue reported on code.google.com by `JustinL...@gmail.com` on 18 Feb 2010 at 8:54

**Comment by GoogleCodeExporter** [comment]
```
In this context you can use both "GETSET" setting 0 as value, or to handle a 
more 
general atomic Get and Del is possible to use this pattern:

temp_id = INCR temp.id
RENAME mykey tempkey.$temp_id
GET tempkey.$temp_id
DEL tempkey.$temp_id
```

Original comment by `anti...@gmail.com` on 18 Feb 2010 at 10:55

**Comment by GoogleCodeExporter** [comment]
Original comment by `anti...@gmail.com` on 27 Feb 2010 at 11:55
- Changed state: **WontFix**
